### ISSUE TITLE
Improve error for Balance in place of U128 

### ISSUE THREAD

**Issue by willemneal** [near/near-sdk-rs#546]
I just ran into this error:

```
Failure [tenk.test.near]: Error: {"index":4,"kind":{"ExecutionError":"Smart contract panicked: panicked at 'Failed to deserialize input from JSON.: Error(\"invalid number\", line: 1, column: 170)', contracts/tenk/src/lib.rs:67:1"}}
```

I had absentmindedly used Balance in place of U128.  Perhaps we should fail to compile if there is a `Balance` in any public parameter.  

Or is it worth wrapping `u128` like `Gas` so that we could use `Balance` and attach a helper implementation?

**Comment by austinabell** [comment]
I guess this follows on some similar discussion around if we should have `Balance` be a wrapper type. I am very hesitant since there are a lot of arithmetic operations that would be difficult to keep up with what's implemented by default for `u128`, but maybe it's less of a negative than this foot-gun of integer vs string serialization.

The counterpoint to this is that there is nothing on Rust or JSON that limits u128/u64 from being serialized as a number not a string and is only a limitation on the JS side. Feels weird to create explicit support for external usage when someone may expect or want the serialization to be an integer and not a string. I don't understand why a large integer could not be parsed into a big integer inside JS, for example.

**Comment by willemneal** [comment]
I agree that JSON should be able to handle integers larger than `2^53 - 1`, which is JS's implementation's limit because their number primitive is a f64.   `BigInt` or `bigint` has been added to JS and is widely supported in browsers and in node. However, JSON seems to lag behind as it doesn't use this.

Here is one library I found for JS that might help with this: https://github.com/sidorares/json-bigint so that we could move to just having bigints as decimals and not strings.

However, as I've been working with @chadoh's new [near-units](https://github.com/near/units-js) library, it would be very useful to have a better interface for dealing with large units with different bases, e.g. NEAR vs wETH.

So while I agree that the implementation of adding all of the needed arithmetic operations, if we can do it once we can provide an easy way to implement it again for other types.

**Comment by r-near** [comment]
Resolved — the `Balance` alias has been removed; `NearToken` is the recommended strong newtype, eliminating the original `u128` foot-gun.
